src/Preg.php

```diff
@@ -35,6 +64,8 @@ final class Preg
      */
     public static function match($pattern, $subject, &$matches = null, $flags = 0, $offset = 0)
     {
+        $pattern = (string) self::verify($pattern);
```

Comment:
```suggestion
        self::verify($pattern);
```

Reply:
Wouldn't that ⬆️ be enough?

Reply:
@kubawerlos If you have static code analysis turned on in the IDE, you will get the notice that it's ambiguous (array is not a string). The cast to string makes the intended use clear (and runtime will give any nice array to string conversion warning if used in error).

Reply:
@ktomk so static code analysis turned on in the IDE would have problem that `$pattern` can be a string, but not that "cannot cast array<string>|string to string" (this is what PHPStan reports on that casting)? Not the best code analysis IMHO.

But that's not what I meant here - I've suggested to remove the assignment as the return value would always be equal to argument.

Reply:
@kubawerlos The static analysis I was referring to is Phpstorm EAP w/ EA Inspections, I introduced it in there b/c I got it marked. I personally have no problem to not have it, I thought it might be useful.

And please excuse my ignorance, I have not seen that Phpstan is available in the project, it should be no problem for me to have it run before pushing commits. If you can help me getting a quick start like which script to call in the project, would be great.

Reply:
It's not in project yet, currently on PR, I remembered about this as I have PHPStan complaining about the casting a lot in some legacy project I was adding it to - for me personally casting is a last resort - if I cannot make it better.

Reply:
@kubawerlos You raise a valid point here, just tried it out w/ Phpstan in the project configuration. It needs level 6 to trigger the issue and the overall code-base is full of it. Let the *Preg* class alone. This requires so much rewriting I'm pretty sure this is out of scope for this PR. However as I have it already running, let me know if you like to have such kind of changes in this PR, this is pretty straight forward to do w/ Phpstan, so just let me know.

Reply:
@kubawerlos I just update the PR to have phpstan on level 7 (max AFAIK) pass for the new lines of code. This resolves the string cast issue. For the benefit of it only requires an array cast at the other places. Good hint here.

Still interested if you want to see fixes based on Phpstan error reporting for the overall PR, as written this is easy to file, but I want to know of this is really needed for acceptance here.

Reply:
The verify method is gone for the current changes. I assume this addresses your comments.